### PR TITLE
Add isRelative to effects that it makes sense for

### LINKED ISSUE #398: Relative or absolute argument on effects
Have an argument which decides whether the effect should be relative or absolute.
For example, if a component's effect has size: 10 as an argument, should the component be resized to 10 (absolute) or add 10 to the size of the component (relative).

### PR DESCRIPTION
**Pull request by spydon**
# Description

Gives the user the ability to define whether an effect should be relative or not.

Fixes #398

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] I have made corresponding changes to the documentation
- [x] I have added examples for new features in `doc/examples`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: lib/effects/move_effect.dart [hunk 1]
```diff
@@ -1,62 +1,42 @@
...
+  Vector2 _startPosition;
+  Vector2 _delta;
 
   MoveEffect({
     @required this.destination,
     @required this.speed,
     this.curve,
     isInfinite = false,
     isAlternating = false,
+    isRelative = true,
```
Comment (spydon) [inline lib/effects/move_effect.dart:20]: Should be false by default.

Comment (spydon) [reply lib/effects/move_effect.dart]: Thanks, fixed

File: lib/effects/scale_effect.dart [hunk 2]
```diff
@@ -4,51 +4,39 @@ import 'package:meta/meta.dart';
...
   ScaleEffect({
     @required this.size,
     @required this.speed,
     this.curve,
     isInfinite = false,
     isAlternating = false,
+    isRelative = false,
     Function onComplete,
-  }) : super(isInfinite, isAlternating, onComplete: onComplete);
+  }) : super(isInfinite, isAlternating,
```
Comment (luanpotter) [inline lib/effects/scale_effect.dart:22]: one per line with trailing commas or all in one line :p

File: lib/effects/rotate_effect.dart [hunk 3]
```diff
@@ -7,36 +7,35 @@ class RotateEffect extends PositionComponentEffect {
...
   RotateEffect({
     @required this.radians, // As many radians as you want to rotate
     @required this.speed, // In radians per second
     this.curve,
     isInfinite = false,
     isAlternating = false,
+    isRelative = false,
     Function onComplete,
-  }) : super(isInfinite, isAlternating, onComplete: onComplete);
+  }) : super(isInfinite, isAlternating,
```
Comment (luanpotter) [inline lib/effects/rotate_effect.dart:21]: ditto

File: lib/effects/move_effect.dart [hunk 4]
```diff
@@ -1,62 +1,42 @@
...
   MoveEffect({
     @required this.destination,
     @required this.speed,
     this.curve,
     isInfinite = false,
     isAlternating = false,
+    isRelative = false,
     Function onComplete,
-  }) : super(isInfinite, isAlternating, onComplete: onComplete);
+  }) : super(isInfinite, isAlternating,
```
Comment (luanpotter) [inline lib/effects/move_effect.dart:22]: ditto

**Review by luanpotter — APPROVED** [review]
LGTM with minor formatting nit. btw, can this be on v0? seems like a non-breaking change that we could just merge on develop instead of v1. but if there are complications I am ok with v1 as well, just curious.

**Comment by spydon** [comment]
> LGTM with minor formatting nit. btw, can this be on v0? seems like a non-breaking change that we could just merge on develop instead of v1. but if there are complications I am ok with v1 as well, just curious. 

It is a breaking change unfortunately, because the rotational one used to be relative and now all are absolute by default. And they use the vector2 now.

File: lib/effects/scale_effect.dart (hunk 2 above)
Comment (spydon) [reply lib/effects/scale_effect.dart]: Good catch, fixed!

**Comment by spydon** [comment]
> > LGTM with minor formatting nit. btw, can this be on v0? seems like a non-breaking change that we could just merge on develop instead of v1. but if there are complications I am ok with v1 as well, just curious.
> 
> It is a breaking change unfortunately, because the rotational one used to be relative and now all are absolute by default. And they use the vector2 now.

Scratch that, it is only internal and I think that the fix with relative -> absolute for the RotateEffect can be seen as a bug fix. So I'll point it to develop.

**Review by erickzanardo — APPROVED** [review]

**Review by renancaraujo — APPROVED** [review]

**Review by luanpotter — APPROVED** [review]
LGTM

**Pull request merged**